### DISCUSSION TITLE
add slash command

### DISCUSSION THREAD

**Discussion by doritogon303** [jagrosh/MusicBot#635] [Ideas]
add slash command

**Comment by DemonFiend** [comment]
This is already easily implemented, in the config i recommend using assign `altprefix = "/"` &` prefix = "@mention"` and allow slash commands in your discord channel. Works like a charm on my server and allows for you to still have a 2nd means to control the bot with the @ ability.

**Reply by eltaylor1104** [reply]
> This is already easily implemented, in the config i recommend using assign `altprefix = "/"` &` prefix = "@mention"` and allow slash commands in your discord channel. Works like a charm on my server and allows for you to still have a 2nd means to control the bot with the @ ability.

That doesn't work for me, the prefix works, but the slash command feature does not. I know they are enabled in the channel, I'm just not sure that they are natively supported in MusicBot yet.

**Reply by jagrosh** [reply]
> > This is already easily implemented, in the config i recommend using assign `altprefix = "/"` &` prefix = "@mention"` and allow slash commands in your discord channel. Works like a charm on my server and allows for you to still have a 2nd means to control the bot with the @ ability.
> 
> That doesn't work for me, the prefix works, but the slash command feature does not. I know they are enabled in the channel, I'm just not sure that they are natively supported in MusicBot yet.

Slash commands haven't been implemented yet

**Reply by doritogon303** [reply]
> > > This is already easily implemented, in the config i recommend using assign `altprefix = "/"` &` prefix = "@mention"` and allow slash commands in your discord channel. Works like a charm on my server and allows for you to still have a 2nd means to control the bot with the @ ability.
> > 
> > 
> > That doesn't work for me, the prefix works, but the slash command feature does not. I know they are enabled in the channel, I'm just not sure that they are natively supported in MusicBot yet.
> 
> Slash commands haven't been implemented yet

will be it added?

**Reply by jagrosh** [reply]
Slash commands will probably be implemented eventually, but it's a bit tricky since they need to be registered in a separate manner. Additionally, other bots running on the same bot account (which is not recommended, but some people end up doing it anyway) could mess with the available slash commands, leading to a confusing or inconsistent experience.

**Comment by DemonFiend** [comment]
I would just like to point this out.
![image](https://user-images.githubusercontent.com/32527478/113067102-2069f600-9192-11eb-83a4-7dd1f774ec4a.png)

**Reply by jagrosh** [reply]
That isn't using slash commands, that is just using `/` as a prefix

**Reply by doritogon303** [reply]
> I would just like to point this out.

this is slash commands 
![Discord_hYaFjWMswk](https://user-images.githubusercontent.com/46861834/113159192-148b3c00-9245-11eb-86a9-1ed6df1d701f.png)

**Comment by NiwiTheFox** [comment]
Hello
I think JDA supports discord slash command already
Just want ask when it will be implemented in jmusicbot
Just curious if there´s some ETA planned

**Reply by MichailiK** [reply]
JDA supports slash commands, but JDA-Utilities (which handles all commands on JMusicBot) does not yet.

**Reply by NiwiTheFox** [reply]
ohhh i thought JDA and JDA-Utils are same
my mistake then

thx

**Reply by silasary** [reply]
From the looks of it, JDA-Utils is abandoned, but this fork https://github.com/Chew/JDA-Chewtils has added slash commands

**Reply by jagrosh** [reply]
It isn't abandoned, I have a local branch where I am implementing the JDA5 changes

**Comment by userzfr** [comment]
Slash commande code in discord.js

> const lib = require('lib')({token: process.env.STDLIB_SECRET_TOKEN});
> 
> await lib.discord.commands['@0.0.0'].create({
>   "name": "play",
>   "description": "Play a music in a bot",
>   "options": [
>     {
>       "type": 3,
>       "name": "name",
>       "description": "Name of song or URL",
>       "required": true
>     }
>   ]
> });
